profiles/common/modules/features/nexteuropa_dgt_connector/ne_tmgmt_dgt_ftt_translator/src/Tools/DataProcessor.php

```diff
@@ -374,22 +358,85 @@ private function getRequestIdentifierDefaults(TMGMTJob $job) {
     );
   }
 
+  /**
+   * Provides the latest DGT FTT Translator Mapping entity for a given code.
+   *
+   * @param string $requester_code
+   *   Requester code.
+   *
+   * @return \Drupal\ne_tmgmt_dgt_ftt_translator\Entity\DgtFttTranslatorMapping | bool
+   *   Entity or FALSE if there are no entries in the entity table.
+   */
+  private function getLatestMappingByRequesterCode($requester_code) {
+    // Querying for the latest mapping for a given requester code.
+    $latest_entity_id = db_query('
```

Comment:
Hi @netlooker , I don't think the QA team will be happy with a db_query.  I think you should use db_select

Reply:
The previous release based on db_query passed and as this query would be used all the time I've decided to use it because of the performance constraints. It's just 3 times faster.

Reply:
https://www.drupal.org/project/drupal/issues/1881146#comment-6907576